### PR TITLE
Bratseth/healt check improvements

### PR DESCRIPTION
**Pull request by bratseth**
@baldersheim see the commit message on the first commit for an explanation

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java [hunk 1]
```diff
@@ -37,11 +39,15 @@ public ClustersStatus() { }
...
     private final Map<String, Boolean> clusterStatus = new HashMap<>();
 
-    public void setContainerHasClusters(boolean containerHasClusters) {
+    /** Sets the current clusters of this container */
+    public void setClusters(Set<String> clusters) {
         synchronized (mutex) {
-            this.containerHasClusters = containerHasClusters;
-            if ( ! containerHasClusters)
-                clusterStatus.clear(); // forget container clusters which was configured away
+            this.containerHasClusters = clusters.size() > 0;
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java:45]: Use isEmpty ?

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java [hunk 2]
```diff
@@ -423,21 +423,26 @@ public boolean isClosed() {
...
         startConfigThread(() -> {
                 while (!isClosed()) {
                     try {
-                        if (nextConfig()) {
-                            if (handle.isChanged()) singleSubscriber.configure(handle.getConfig());
-                        }
-                    } catch (Exception e) {
-                        log.log(SEVERE, "Exception from config system, continuing config thread: " + Exceptions.toMessageString(e));
+                        if (nextConfig() && handle.isChanged())
+                            singleSubscriber.configure(handle.getConfig());
```
Comment (baldersheim) [inline config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java:440]: You probably need a separate catch for nextConfig() that is SEVERE to ensure you you only blame the subscriber when it is truly to blame.

**Review by baldersheim — APPROVED** [review]

File: container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java (hunk 1 above)
Comment (bratseth) [reply container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java:45]: I actually think this is slightly clearer by avoiding negation.

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java (hunk 2 above)
Comment (bratseth) [reply config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java:449]: Good point, done.

**Pull request merged**